### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by Arturgouveia1970**
In this PR:
- Installed devise gem
- Configured devise

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
```
Comment (medaminedev66) [inline app/views/devise/registrations/new.html.erb:27]: - [ ] I'm not able to sign up and create a User using this form. I examined the User model and found validation for the `:name` (presence: true). I recommend adding a field for the `:name` in your form so the user is able to add a name and sign up in your app.
There is another validation for `:posts_counter` in the User model. Please take it into consideration and ensure that the user is able to register for your app.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (medaminedev66) [inline app/models/user.rb:5]: - [ ] Following the project requirements, the application asks for confirmation of email. I recommend adding `:confirmable` to the devise:
    ```ruby
    	devise :database_authenticatable, :registerable,
    				:recoverable, :rememberable, :validatable, :confirmable
    ````

File: db/migrate/20220907103047_add_devise_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (medaminedev66) [inline db/migrate/20220907103047_add_devise_to_users.rb:28]: - [ ] After adding `:confimable` to the devise in the User model, don't forget to uncomment the Confirmable section in your migration.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @Arturgouveia1970  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Devise has been installed and setup
- You have a descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (Whoistolu) [reply app/views/devise/registrations/new.html.erb]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @Arturgouveia1970 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**